packages/node_modules/@webex/internal-plugin-device/test/unit/spec/interceptors/device-url.js

```diff
@@ -73,29 +73,20 @@ describe('plugin-device', () => {
         });
 
         describe('when service does not exist', () => {
-          it('should return the options unchanged when waitForService returns undefined', () => {
+          beforeEach('setup device url, options, and stubs', () => {
             interceptor.webex.internal.services.waitForService = sinon.stub().resolves();
-            interceptor.webex.internal.device.url = undefined;
-            options = {
-              headers: {
-                'cisco-device-url': undefined
-              },
-              ...options
-            };
+            interceptor.webex.internal.device.url = 'http://device-url.com/';
+          });
+
+          it('should return the options unchanged when getServiceFromUrl returns undefined', () => {
+            interceptor.webex.internal.services.getServiceFromUrl = sinon.stub().returns();
 
             return interceptor.onRequest({...options})
               .then((results) => assert.deepEqual(results, options));
           });
 
-          it('should return the options unchanged when waitForService returns an empty object', () => {
-            interceptor.webex.internal.services.waitForService = sinon.stub().resolves({});
-            interceptor.webex.internal.device.url = undefined;
-            options = {
-              headers: {
-                'cisco-device-url': undefined
-              },
-              ...options
-            };
+          it('should return the options unchanged when getServiceFromUrl returns a URL that is not in the catalog', () => {
+            interceptor.webex.internal.services.getServiceFromUrl = sinon.stub().returns('http://not-in-catalog.com/');
```

Comment:
getServiceFromUrl should return an object, correct?

Reply:
Addressed this by refactoring the tests.